src/main/java/Task.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Class that represents the task.
+ * Contains taskName, done and taskType.
+ */
+public abstract class Task {
+
+    private String taskName; //name of the given task
+    private boolean isDone; //whether the task is done or not
+    private String taskType;
+
+    /**
+     * Constructor.
+     * @param taskName name of task.
+     * @param taskType what type is it.
+     */
+    public Task(String taskName, String taskType) {
+        this.taskName = taskName;
+        this.isDone = false;
+        this.taskType = taskType;
+    }
+
+    /**
+     * Function to set the task to done.
+     */
+    public void setDone() {
+        this.isDone = true;
+    }
+
+
+    /**
+     * Function to get the task current status - done or not done.
```

Comment:
In method header comments, the first sentence should start in the form Returns ..., Sends ..., Adds ... (not Return or Returnning etc.). Instead of starting with "Function...", you could start with the description of what the method is doing (i.e. "Gets the..."). There should also be an empty line between description and parameter section.

This applies to all methods in this class.